File: tests/functional_tests/src/hurdle_en.py
# General solution to hurdle worlds
think(0)

def turn_right():
    repeat 3:
        turn_left()

def follow_right_wall():
    if right_is_clear():
        turn_right()
        move()
    # let's use two equivalent tests
    elif front_is_clear() and not wall_in_front():
        move()
    else:
        turn_left()

while not at_goal():
    follow_right_wall()

File: tests/integration_tests/tutorial_en/programs/hurdle_en.js
// overkill - single program for Hurdle 1, 2, 3, 4
function right() {
    turn_left();
    turn_left();
    turn_left();
}

function follow_wall () {
    if (right_is_clear()) {
        right();
        move();
    } else if (front_is_clear() && !wall_in_front()) { // two equivalent tests
        move();
    } else {
        turn_left();
    }
}

while (! at_goal()) {
    follow_wall();
}
